staging/vhost-device-sound/src/audio_backends/pipewire/test_utils.rs

```diff
@@ -0,0 +1,129 @@
+// Manos Pitsidianakis <manos.pitsidianakis@linaro.org>
+// SPDX-License-Identifier: Apache-2.0 or BSD-3-Clause
+
+use std::{
+    path::Path,
+    process::{Child, Command},
+    sync::OnceLock,
+};
+
+use tempfile::{tempdir, TempDir};
+
+/// We want all threads of the same process to share this harness, because cargo
+/// building will block other cargo processes. So build a harness once and share
+/// it with other tests.
+static CELL: OnceLock<PipewireTestHarness> = OnceLock::new();
```

Comment:
Since it is static, when are the destructors invoked?

Reply:
Yeah that's why pkill etc aren't called. Will rewrite it

Reply:
If it's too complicated to implement a singleton, I think it's okay to launch multiple demons (one per test). Or have you already had problems with that?

Reply:
Well I goofed it up yesterday because I was working fast and carelessly. I was at the time also building the vhost-device-sound binary, so I needed a singleton because the cargo builds were blocking each other. But I hadn't realised that the binary was not needed anymore since the test became a utility module for the tests in pipewire.rs.